src/xrpld/app/misc/NetworkOPs.cpp

```diff
@@ -1229,15 +1239,25 @@ NetworkOPsImp::processTransaction(
         return;
     }
 
+    auto const view = m_ledgerMaster.getCurrentLedger();
+
+    // This function is called by several different parts of the codebase
+    // under no circumstances will we ever accept an inner txn within a batch
+    // txn from the network.
+    auto const sttx = *transaction->getSTransaction();
+    if (sttx.isFlag(tfInnerBatchTxn) && view->rules().enabled(featureBatch))
+    {
```

Comment:
Is it possible to write a unit test for this?

Reply:
added: https://github.com/XRPLF/rippled/pull/5060/commits/569ac2102e40da71d0e5216c50985f3c29cc6f67